Fabulous.XamarinForms/src/Fabulous.XamarinForms.Core/ViewUpdaters.fs

```diff
@@ -38,31 +38,44 @@ module ViewUpdaters =
            (create: 'T -> 'TargetT)
            (attach: 'T voption -> 'T -> 'TargetT -> unit) // adjust attached properties
            (canReuse : 'T -> 'T -> bool) // Used to check if reuse is possible
+           (getKey:'T->string)
            (update: 'T -> 'T -> 'TargetT -> unit) // Incremental element-wise update, only if element reuse is allowed
         =
         match prevCollOpt, collOpt with 
         | ValueSome prevColl, ValueSome newColl when identical prevColl newColl -> ()
         | _, ValueNone -> targetColl.Clear()
         | _, ValueSome coll ->
-            if (coll = null || coll.Length = 0) then
-                targetColl.Clear()
-            else
+            
+
+                let prevColl =
+                    match prevCollOpt with
+                    | ValueSome x -> x
+                    | _ -> [||]
+    
                 // Remove the excess targetColl
-                while (targetColl.Count > coll.Length) do
-                    targetColl.RemoveAt (targetColl.Count - 1)
+
 
                 // Count the existing targetColl
                 // Unused variable n' introduced as a temporary workaround for https://github.com/fsprojects/Fabulous/issues/343
                 let _ = targetColl.Count
                 let n = targetColl.Count
-
+                let targetControls = Dictionary<string,'TargetT>()
+                for i in 0..prevColl.Length-1 do
+                        let key = getKey prevColl.[i]
+                        targetControls.[key]<-targetColl.[i]
+                         
+                while (targetColl.Count > coll.Length) do
+                  targetColl.RemoveAt (targetColl.Count - 1)
```

Comment:
Doing that, there's a chance you're deleting controls you might reuse.

Example:
```fs
// Previous
[
    View.Label(text = "Label")
    View.Button(key = "MyButton", text = "Button")
]

// Current
[
    View.Button(key = "MyButton", text = "SuperButton")
]
```

This will result in removing the button since it's outside of the new range.
But then you'll try to update it since the key is matching, even though it disappeared from the screen.

Reply:
This will not happen, because i have separate dictionary for such controls. targetColl is not a place, where we seek for controls after we put them in dictionary.

Reply:
It was an issue previously because the controls in excess were removed from the list Xamarin.Forms was using (e.g. ListView's `Items` or StackLayout's `Children`).
Even though they were kept in an another list so they could be reused, they were not reinserted to the Xamarin.Forms's list, removing them from the UI incorrectly.

But now, it's no longer an issue since you set all items an index.